android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -263,6 +285,36 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
     }
   }
 
+  suspend fun setInitialExpression(questionnaire: Questionnaire) {
+    // TODO handle hierarchy and scope for variable items
+    // TODO add functionality to SDK instead
+    questionnaire.item.flattened().forEach { item ->
+      item
+        .initialExpression
+        ?.takeIf { it.language == "application/x-fhir-query" }
+        ?.let { expression ->
+          val answerOptions: List<QuestionnaireItemAnswerOptionComponent> =
+            QuestionnaireFragment::class.callSuspendFunctionOnField(
```

Comment:
I think it's best to expose this function from the SDK instead of using Reflection to invoke the function. You can submit a PR and have it in the next release of the SDC library.

cc @ndegwamartin

Reply:
Yeah. I am creating the issues and PRs on SDK as well. This is supposed to be changed whenever SDK has functionality available. There was no way to implement this without heavily coping functions and extension from SDK. This involved multiple classes and methods which would have become difficult to manage overtime.

Reply:
@ndegwamartin Can we have a preview release with the linked PRs on this issue?

Reply:
I'm equally okay with merging the PR with the reflection code and applying the refactor later on.